core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -4574,26 +4625,34 @@ private void analyzeRowFilter(String currentIdentity, Table table, QualifiedObje
                         correlationSupport);
             }
             catch (TrinoException e) {
-                throw new TrinoException(e::getErrorCode, extractLocation(table), format("Invalid row filter for '%s': %s", name, e.getRawMessage()), e);
-            }
-            finally {
-                analysis.unregisterTableForRowFiltering(name, currentIdentity);
+                throw new TrinoException(e::getErrorCode, extractLocation(table), format("%s for '%s': %s", message, name, e.getRawMessage()), e);
             }
+        }
 
-            analysis.recordSubqueries(expression, expressionAnalysis);
-
+        private void addCoercionForRowConstraint(Table table, QualifiedObjectName name, Expression expression, ExpressionAnalysis expressionAnalysis, String constraintType)
```

Comment:
There's nothing special about a "row constraint" in this method. If anything, it should be called `coerceToBoolean`, but it also feels like spurious reuse of code.